src/codewhisperer/util/telemetryHelper.ts

```diff
@@ -178,7 +178,7 @@ export class TelemetryHelper {
                 recommendationSuggestionState?.set(i, 'Empty')
             }
             const event: CodewhispererUserDecision = {
-                codewhispererRequestId: requestId,
+                codewhispererRequestId: requestIdList[i],
```

Comment:
Ideally it should be a list of recommenationContexts where in each context there's a recommendation, and a requestId(and other things related to that completion). So you don't have to maintain two separate list and worry about the potential problem of inconsistency between these two lists.

Reply:
What if there is no recommendation? but still we need to send the requestId? `recordUserDecisionTelemetryForEmptyList`. I think, in this case we can not map recommendation to requestId.
Correct me if I am wrong?

Reply:
for that one you can just use the local requestId and pass that to the event, for the rest, we can add the recommendationContext to the session using similar logic in line 383 of the file recommendationHandler.ts

Reply:
This is not a blocker for now.

Reply:
Added a TODO for further changes.